translator/trace/protospan_to_jaegerthrift.go

```diff
@@ -0,0 +1,381 @@
+				Key:   k,
+				VType: jaeger.TagType_STRING,
+				VStr:  &str,
+			}
+			jTags = append(jTags, jTag)
+		}
+	}
+
+	if node.Identifier != nil {
+		if node.Identifier.HostName != "" {
+			hostTag := &jaeger.Tag{
+				Key:   "hostname",
+				VType: jaeger.TagType_STRING,
+				VStr:  &node.Identifier.HostName,
+			}
+			jTags = append(jTags, hostTag)
+		}
+		if node.Identifier.Pid != 0 {
+			pid := int64(node.Identifier.Pid)
+			hostTag := &jaeger.Tag{
+				Key:   "pid",
+				VType: jaeger.TagType_LONG,
+				VLong: &pid,
+			}
+			jTags = append(jTags, hostTag)
+		}
+		if node.Identifier.StartTimestamp != nil && node.Identifier.StartTimestamp.Seconds != 0 {
+			startTimeStr := ptypes.TimestampString(node.Identifier.StartTimestamp)
+			hostTag := &jaeger.Tag{
+				Key:   "start.time",
+				VType: jaeger.TagType_STRING,
+				VStr:  &startTimeStr,
+			}
+			jTags = append(jTags, hostTag)
+		}
+	}
+
+	// Add OpenCensus library information as tags if available
+	ocLib := node.LibraryInfo
+	if ocLib != nil {
+		// Only add language if specified
+		if ocLib.Language != commonpb.LibraryInfo_LANGUAGE_UNSPECIFIED {
+			languageStr := ocLib.Language.String()
+			languageTag := &jaeger.Tag{
+				Key:   "opencensus.language",
+				VType: jaeger.TagType_STRING,
+				VStr:  &languageStr,
+			}
+			jTags = append(jTags, languageTag)
+		}
+		if ocLib.ExporterVersion != "" {
+			exporterTag := &jaeger.Tag{
+				Key:   "opencensus.exporterversion",
+				VType: jaeger.TagType_STRING,
+				VStr:  &ocLib.ExporterVersion,
+			}
+			jTags = append(jTags, exporterTag)
+		}
+		if ocLib.CoreLibraryVersion != "" {
+			exporterTag := &jaeger.Tag{
+				Key:   "opencensus.corelibversion",
+				VType: jaeger.TagType_STRING,
+				VStr:  &ocLib.CoreLibraryVersion,
+			}
+			jTags = append(jTags, exporterTag)
+		}
+	}
+
+	var serviceName string
+	if node.ServiceInfo != nil && node.ServiceInfo.Name != "" {
+		serviceName = node.ServiceInfo.Name
+	}
+
+	if serviceName == "" && len(jTags) == 0 {
+		// No info to put in the process...
+		return nil
+	}
+
+	jProc := &jaeger.Process{
+		ServiceName: serviceName,
+		Tags:        jTags,
+	}
+
+	return jProc
+}
+
+func ocSpansToJaegerSpans(ocSpans []*tracepb.Span) ([]*jaeger.Span, error) {
+	if ocSpans == nil {
+		return nil, nil
+	}
+
+	// Pre-allocate assuming that few, if any spans, are nil.
+	jSpans := make([]*jaeger.Span, 0, len(ocSpans))
+	for _, ocSpan := range ocSpans {
+		traceIDLow, traceIDHigh, err := traceIDBytesToLowAndHigh(ocSpan.TraceId)
+		if err != nil {
+			return nil, fmt.Errorf("OC span has invalid trace ID: %v", err)
+		}
+		jReferences, err := ocLinksToJaegerReferences(ocSpan.Links)
+		if err != nil {
+			return nil, fmt.Errorf("Error converting OC links to Jaeger references: %v", err)
+		}
+		spanID, err := ocIDBytesToJaegerID(ocSpan.SpanId)
+		if err != nil {
+			return nil, fmt.Errorf("OC span has invalid span ID: %v", err)
+		}
+		parentSpanID, err := ocIDBytesToJaegerID(ocSpan.ParentSpanId)
```

Comment:
Since there will be special cases for the IDs on the function that understand the difference between ParentSpanId, SpanId and so on, I will do a bit different but overall with same intended effect.